build/gen_stub.php

```diff
@@ -1047,16 +1047,36 @@ function (Expr $expr) use (&$defaultValueConstant) {
         if ($this->type) {
             $arginfoType = $this->type->toArginfoType();
             if ($arginfoType->hasClassType()) {
-                $simpleType = $this->type->tryToSimpleType();
-
-                $className = $arginfoType->classTypes[0]->name;
-                $code .= "	zend_string *property_{$propertyName}_class_{$className} = zend_string_init(\"$className\", sizeof(\"$className\")-1, 1);\n";
-                if ($simpleType) {
-                    $typeCode = "(zend_type) ZEND_TYPE_INIT_CLASS(property_{$propertyName}_class_{$className}, " .  ((int) $this->type->isNullable()) . ", 0)";
-                } elseif (count($arginfoType->classTypes) === 1) {
-                    $typeCode = "(zend_type) ZEND_TYPE_INIT_CLASS(property_{$propertyName}_class_{$className}, 0, " . $arginfoType->toTypeMask() . ")";
+                if (count($arginfoType->classTypes) >= 2) {
+                    foreach ($arginfoType->classTypes as $classType) {
+                        $className = $classType->name;
+                        $code .= "\tzend_string *property_{$propertyName}_class_{$className} = zend_string_init(\"$className\", sizeof(\"$className\") - 1, 1);\n";
+                    }
+
+                    $classTypeCount = count($arginfoType->classTypes);
+                    $code .= "\tzend_type_list *property_{$propertyName}_type_list = malloc(ZEND_TYPE_LIST_SIZE($classTypeCount));\n";
+                    $code .= "\tproperty_{$propertyName}_type_list->num_types = $classTypeCount;\n";
+
+                    foreach ($arginfoType->classTypes as $k => $classType) {
+                        $className = $classType->name;
+                        $code .= "\tproperty_{$propertyName}_type_list->types[$k] = (zend_type) ZEND_TYPE_INIT_CLASS(property_{$propertyName}_class_{$className}, 0, 0);\n";
+                    }
+
+                    $isNullable = $this->type->isNullable();
+
+                    $code .= "\tzend_type property_{$propertyName}_type = ZEND_TYPE_INIT_PTR(property_{$propertyName}_type_list, _ZEND_TYPE_LIST_BIT, $isNullable, 0);\n";
```

Comment:
I have a problem here: I don't understand how I could add support for types like `Foo|Bar|int|null`. I tried to add the simple type mask as the last parameter of `ZEND_TYPE_INIT_PTR()` as well as appending it to the type list, but neither worked for me.

Reply:
I would expect `0, toTypeMask()` to work.

Reply:
Indeed! What I messed up with is that I tried to pass `(zend_type) ZEND_TYPE_INIT_MASK(" . $arginfoType->toTypeMask() . ")` (which was copy-pasted from a few lines below) as the last parameter rather than just the type mask.

Reply:
Additionally, I committed the other changes.